tests/conftest.py

```diff
@@ -75,6 +75,12 @@ def tls_certificate_pem_path(tls_certificate):
         yield cert_pem
 
 
+@pytest.fixture
+def tls_certificate_private_key_path(tls_certificate_authority):
```

Comment:
Could you move this closer to `tls_ca_certificate_pem_path`? And maybe use a fixture name that is similar. Like `tls_ca_private_key_pem_path`.